### PR TITLE
fix: pass kwargs `env` to both update and test targets

### LINKED ISSUE #2270: regression: `pip_compile` ignores `env` attribute
# 🐞 bug report

### Affected Rule

The issue is caused by the rule: `pip_compile` but exposed to public as `compile_pip_requirements`

During this change below, `env` parameter used to be captured by `kwargs` and then passed as `attrs` to respective `py_binary`. But now `env` within `kwargs` is being ignored, so any environment variables that is set with `pip_compile` rule is being ignored.

https://github.com/bazelbuild/rules_python/commit/e923f9e86c29dde355ffbef80420e5de843a79e7#diff-a92cc6468f47191b8e22d4bbe2564d3968d3dd4bbc383b9f3df3c4c520e1583eL147-R165

### Is this a regression?

Yes. `env` parameter that was passed to `pip_compile` was respected in `0.34.0` but now it's not on `0.36.0`.

### Description

 A clear and concise description of the problem...

## 🔬 Minimal Reproduction

```
compile_pip_requirements(
    ...,
    env = {
        "PIP_CONFIG_FILE": "<path>",
    },
    ...
)
```

Then run `bazel run :requirements.update` on `0.34.0` vs `0.36.0`

## 🔥 Exception or Error

</code></pre>

No particular exception, but the environment variables are not picked up when running `bazel run :requirements.update`

## 🌍  Your Environment

**Operating System:** MacOS 14.7

**Output of `bazel version`:** bazel 6.5.0

**Rules_python version:** `0.34.0` trying to upgrade to `0.36.0`

### PR DESCRIPTION
**Pull request by lpulley**
Fixes #2270.

### CODE REVIEW

**Comment by cj81499** [comment]
_Seems_ reasonable to me. If it's not too much a hassle, it'd be good to write an automated test for this to avoid future breakage.

**Review by aignas — COMMENTED** [review]
Could you please add a CHANGELOG note to mentioned what was fixed?

**Comment by lpulley** [comment]
> Could you please add a CHANGELOG note to mentioned what was fixed?

Done!

File: python/private/pypi/pip_compile.bzl [hunk 1]
```diff
@@ -154,18 +154,11 @@ def pip_compile(
...
-    # user-provided configuration.
-    kwargs["env"] = select({
-        "@@platforms//os:windows": {"USERPROFILE": "Z:\\FakeSetuptoolsHomeDirectoryHack"},
-        "//conditions:default": {},
-    }) | kwargs.get("env", {})
+    env = kwargs.pop("env", {})
 
     py_binary(
         name = name + ".update",
+        env = env,
```
[the remark this answers is not in the document]
Comment (lpulley) [reply python/private/pypi/pip_compile.bzl:161]: Well... `X_test` runs as a Bazel action. `X.update` is only expected to be invoked by the user directly, in a normal environment.

[the remark this answers is not in the document]
Comment (cj81499) [reply python/private/pypi/pip_compile.bzl:161]: Seems like a possible source of confusion to have different behavior here, but I trust your (and @aignas's) judgement on this.

**Review by aignas — APPROVED** [review]
Thank you!

**Pull request merged**